pages/_app.tsx

```diff
@@ -135,16 +135,40 @@ export default function PlanetWeb({ Component, pageProps, err }: any) {
 
   const [showVideo, setshowVideo] = React.useState(true);
 
+  // if localShowVideo is undefined
+  // set localShowVideo is true and show the video
+  // if localShowVideo is true show the video
+  // if localShowVideo is false hide the video
+
+  const [localShowVideo, setLocalShowVideo] = React.useState(false);
+
   React.useEffect(() => {
-    if (typeof window !== 'undefined') {
-      if (localStorage.getItem('hidePreview')) {
-        setshowVideo(false);
-      }
-      if (router.pathname !== '/') {
-        setshowVideo(false);
+    if (router.pathname === '/') {
+      if (typeof window !== 'undefined') {
+        if (localStorage.getItem('showVideo')) {
+          if (localStorage.getItem('showVideo') === 'true') {
+            setLocalShowVideo(true);
+          } else {
+            setLocalShowVideo(false);
+          }
+        } else {
+          localStorage.setItem('showVideo', 'true');
+          setLocalShowVideo(true);
+        }
       }
+    } else {
+      setLocalShowVideo(false);
     }
   }, []);
+
+  React.useEffect(() => {
+    if (localShowVideo) {
+      setshowVideo(true);
+    } else {
+      setshowVideo(false);
+    }
```

Comment:
You can directly do setshowVideo(localShowVideo), no need for conditional loops. 

Also if we already have one variable why use new variable `localShowVideo, setLocalShowVideo` ?

Reply:
reduced to setshowVideo(localShowVideo),

To handle the case where component loads before local storage is loaded.